src/metabase/lib/aggregation.cljc

```diff
@@ -64,7 +66,11 @@
      (when base-type
        {:base-type base-type})
      (when effective-type
-       {:effective-type effective-type}))))
+       {:effective-type effective-type})
+     (when agg-name
+       {:name agg-name})
+     (when display-name
+       {:display-name display-name}))))
```

Comment:
This fixes a bug: without these, referencing an alias doesn't work. This might be worth an E2E test too.